src/awkward/_v2/highlevel.py

```diff
@@ -243,6 +243,12 @@ def __init__(
         if with_name is not None:
             layout = ak._v2.operations.with_name(layout, with_name, highlevel=False)
 
+        if (
+            backend is not None
+            and ak._v2._util.regularize_backend(backend) is ak.nplike.Jax.instance()
+        ):
+            layout = ak._v2.packed(layout, highlevel=False)
```

Comment:
I haven't come across a mechanism that would allow us to intercept JAX during it's operation. JAX however will complain as soon as the primals and tangents don't match early on during the execution.

Reply:
Okay, then we shouldn't pack it. Let the user get errors.